packages/react-native/Libraries/Network/RCTNetworking.mm

```diff
@@ -365,6 +365,12 @@ - (RCTURLRequestCancellationBlock)buildRequest:(NSDictionary<NSString *, id> *)q
                                       forHTTPHeaderField:@"Content-Length"];
                                 }
 
+                                // NSRequest default cache policy violate on `If-None-Match`, should allow the request
+                                // to get 304 from server.
+                                if (request.allHTTPHeaderFields[@"If-None-Match"]) {
```

Comment:
This is case sensitive. Will there be cases where these headers come with lower casing?